src/app/games/conspirateurs/tests/ConspirateursRules.spec.ts

```diff
@@ -0,0 +1,522 @@
+/* eslint-disable max-lines-per-function */
+import { Coord } from 'src/app/jscaip/Coord';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { Player } from 'src/app/jscaip/Player';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { RulesUtils } from 'src/app/jscaip/tests/RulesUtils.spec';
+import { ConspirateursFailure } from '../ConspirateursFailure';
+import { ConspirateursMove, ConspirateursMoveDrop, ConspirateursMoveJump, ConspirateursMoveSimple } from '../ConspirateursMove';
+import { ConspirateursNode, ConspirateursRules } from '../ConspirateursRules';
+import { ConspirateursState } from '../ConspirateursState';
+import { ConspirateursMinimax } from '../ConspirateursMinimax';
+
+describe('ConspirateursRules', () => {
+    const _: Player = Player.NONE;
+    const A: Player = Player.ZERO;
+    const B: Player = Player.ONE;
+
+    let rules: ConspirateursRules;
+
+    let minimaxes: Minimax<ConspirateursMove, ConspirateursState>[];
+
+    beforeEach(() => {
+        rules = ConspirateursRules.get();
+        minimaxes = [
+            new ConspirateursMinimax(rules, 'ConspirateursMinimax'),
+        ];
+    });
+
+    function drop(coord: Coord): ConspirateursMove {
+        return ConspirateursMoveDrop.of(coord).get();
+    }
+    function simpleMove(start: Coord, end: Coord): ConspirateursMove {
+        return ConspirateursMoveSimple.of(start, end).get();
+    }
+    function jump(coords: Coord[]): ConspirateursMove {
+        return ConspirateursMoveJump.of(coords).get();
+    }
+    describe('drop moves', () => {
+        it('should allow drops within the center zone', () => {
+            // Given the initial state
+            const state: ConspirateursState = ConspirateursState.getInitialState();
+            // When dropping a piece in the center zone
+            const move: ConspirateursMove = drop(new Coord(7, 7));
+            // Then the move is legal and the piece is put at the expected positionç
```

Comment:
use "Then blblb should krkrkr" not "Then blbl is krkr"
position ne prends pas de cédille :D !

Reply:
changé dans le fichier